test/authenticate.user.test.js

```diff
@@ -0,0 +1,478 @@
+});
+
+/**
+ * Test successful user authentication using a matching email login and password
+ */
+describe("Authenticate user successfully with matching email and password", function() {
+    it("should return: true", function(done) {
+        const
+            email = 'test@dummy.com',
+            password = 'authentication-test';
+        request(app)
+            .post('/api/v1/users')
+            .send({
+                username: 'TestDummy',
+                displayName: 'MostValuedTest',
+                email: email,
+                password: password
+            })
+            .expect(201)
+            .end(function(err, res) {
+                if (err) done(err);
+                authenticateUser({ email: email }, password, function (result) {
+                    assert.equal(result.email === email && result.hashedPassword === password, true);
+                    done();
+                });
+            });
+    });
+});
+
+/**
+ * Test successful user authentication using a matching email login and password
+ */
+describe("Authenticate user successfully with matching username and password", function() {
+    it("should return: true", function(done) {
+        const
+            username = 'TestDummy',
+            password = 'authentication-test';
+        request(app)
+            .post('/api/v1/users')
+            .send({
+                username: username,
+                displayName: 'MostValuedTest',
+                email: 'test@dummy.com',
+                password: password
+            })
+            .expect(201)
+            .end(function(err, res) {
+                if (err) done(err);
+                authenticateUser({ username: username }, password, function (result) {
+                    assert.equal(result.username === username && result.hashedPassword === password, true);
+                    done();
+                });
+            });
+    });
+});
+
+/**
+ * Test successful user authentication using a matching email and username login and password
+ */
+describe("Authenticate user successfully with matching email and username and password", function() {
+    it("should return: true", function(done) {
+        const
+            username = 'TestDummy',
+            email = 'test@dummy.com',
+            password = 'authentication-test';
+        request(app)
+            .post('/api/v1/users')
+            .send({
+                username: username,
+                displayName: 'MostValuedTest',
+                email: email,
+                password: password
+            })
+            .expect(201)
+            .end(function(err, res) {
+                if (err) done(err);
+                authenticateUser({ username: username, email: email }, password, function (result) {
+                    console.log(result._id);
+                    assert.equal(result.email === email &&
+                        result.username === username && result.hashedPassword === password, true);
```

Comment:
This will fail once the hashPassword method is implemented, as result.hashedPassword won't equal password.

Reply:
The feature isn't yet implemented and I would assume whoever is implementing the password hash feature would apply necessary changes here